crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -337,7 +338,11 @@ impl ShaderCache {
                 render_device
                     .wgpu_device()
                     .push_error_scope(wgpu::ErrorFilter::Validation);
-                let shader_module = render_device.create_shader_module(module_descriptor);
+                let shader_module = if unchecked {
+                    unsafe { render_device.create_shader_module_unchecked(module_descriptor) }
```

Comment:
I'll add a SAFETY comment saying that it's safe because the user has confirmed they want to skip shader checks.

Reply:
i don't think that's true if it's opt-out in the material trait.

the wgpu-types approach of making an unsafe function to access the mode is pretty good. so the material trait by default would return `ShaderSafetyMode::safe()` and if users want to disable it they override and return `unsafe { ShaderSafetyMode::unsafe() }` (and you add a private member to the struct so users don't have another way to construct it).

but i wonder what kind of harm omitting the shader bounds checks can cause. maybe the device function shouldn't be "rust" unsafe anyway which is meant for application ub. i think bad shaders might crash the app or return bad data but i don't think they could cause cpu/program/stack memory corruption.

Reply:
Strictly speaking, any safety comment that would be put here is likely insufficient to satisfy the safety invariants of the function call, as we would need to trust the provided shader source code no matter where it came from, including from the disk or over the network, and that is not true in the general case.

It may be worth upstreaming this issue and confirming that unsafe here really means unsafe the way Rust intends the keyword to be used. AFAICT, this omits bounds checks on the GPU, which may result in UB on the GPU or transitively cause transfer the results of UB to be returned from compute shaders.